pleskdistup/actions/common_checks.py

```diff
@@ -142,124 +130,106 @@ def _do_check(self) -> bool:
-            outdated_php_domains = subprocess.check_output(["/usr/sbin/plesk", "db", looking_for_domains_sql_request],
-                                                           universal_newlines=True)
-            outdated_php_domains = [
-                domain[2:-2] for domain in outdated_php_domains.splitlines()
-                if domain.startswith("|") and not domain.startswith("| name ")
-            ]
-            if len(outdated_php_domains) <= 0:
+
+            outdated_php_domains = plesk.get_from_plesk_database(looking_for_domains_sql_request)
+            if not outdated_php_domains:
                 return True
 
             log.debug(f"Outdated PHP domains: {outdated_php_domains}")
             outdated_php_domains = "\n\t- ".join(outdated_php_domains)
-            self.description = self.description.format(self.last_outdated_version, outdated_php_domains)
-
-            return False
-        except Exception:
-            log.error("Unable to get domains list from plesk database!")
+            self.description = self.description.format(
+                modern=self.min_version,
+                domains=outdated_php_domains
+            )
+        except Exception as ex:
+            log.err("Unable to get domains list from plesk database!")
+            raise RuntimeError("Unable to get domains list from plesk database!") from ex
 
-        return True
+        return False
 
 
-class AssertCronUsesOutdatedPHP(action.CheckAction):
-    last_outdated_version: version.PHPVersion
-    name: str
-    description: str
+class AssertMinPHPVersionUsedByCron(action.CheckAction):
+    min_version: version.PHPVersion
 
     def __init__(
-            self,
-            last_outdated_version: str,
+        self,
+        min_version: str,
     ):
         self.name = "checking cronjob uses outdated PHP"
-        self.last_outdated_version = version.PHPVersion(last_outdated_version)
+        self.min_version = version.PHPVersion(min_version)
         self.description = """We have detected that some cronjobs are using outdated PHP versions.
-\tSwitch the following cronjobs to to any version higher than {} in order to continue with the conversion process:"
-\t- {}
+\tSwitch the following cronjobs to {modern} or later in order to continue with the conversion process:"
+\t- {cronjobs}
 
 \tYou can do this in the Plesk web interface by going “Tools & Settings” → “Scheduled Tasks”.
 """
 
     def _do_check(self) -> bool:
-        outdated_php_packages = [
-            f"plesk-php{php.major}{php.minor}" for php in get_known_php_versions() if php < self.last_outdated_version
-        ]
-        log.debug(f"Outdated PHP handlers: {outdated_php_packages}")
+        log.debug(f"Checking for outdated PHP version in cronjobs.  {self.min_version}")
+        if not plesk.is_plesk_database_ready():
+            log.info("Plesk database is not ready. Skipping the outdated php for websites check.")
+            return True
```

Comment:
The conversion process does not rely heavily on the Plesk database. I don't think the distupgrade process does either.
Additionally, I believe the PHP checker is not the appropriate place to check for the availability of the Plesk database. We could include a separate CheckAction specifically for this purpose.

Reply:
Ignoring the check if it can't be performed due to technical problems seems unsafe as default behaviour. By default, the answer must be yes, no or hard fail. I propose to make the soft fail opt-in in such cases, e.g. `AssertMinPhpVersionUsedByCron(optional=True)`.

Reply:
It depends on whether we see the unreachable database as a "technical problem" or not. In the case of conversion/distupgrade, it can be a normal state of the system because the tool disables the Plesk database by itself.
However, I'm okay with it as long as it is optional by default.